crashtracker/src/collector/emitters.rs

```diff
@@ -81,6 +92,66 @@ unsafe fn emit_backtrace_by_frames(
     Ok(())
 }
 
+
+fn emit_backtrace_libuwnind(
+    w: &mut impl Write,
+    ucontext: *const ucontext_t,
+    resolve_symbols: bool,
+) -> anyhow::Result<()> {
+    unsafe {
+        let mut cursor = UnwCursor([0; std::mem::size_of::<UnwCursor>()]);
+        let context = ucontext as *mut UnwContext; // Reinterpret as UnwContext
+
+        writeln!(w, "{DD_CRASHTRACK_BEGIN_STACKTRACE}")?;
+
+        if unw_init_local(&mut cursor, context) != 0 {
+            writeln!(w, "{{ \"error\": \"Failed to initialize cursor from context\" }}")?;
+            writeln!(w, "{DD_CRASHTRACK_END_STACKTRACE}")?;
+            w.flush()?;  
+            return Err(anyhow::anyhow!("Failed to initialize cursor from context"));
+        }
+
+        let mut name_buf = vec![0 as libc::c_char; 256]; // Buffer for function names
+        let mut offset: u64 = 0;
+
+        loop {
+            let mut ip: u64 = 0;
+            let mut sp: u64 = 0;
+
+            if unw_get_reg(&mut cursor, UNW_REG_IP, &mut ip) != 0 ||
+               unw_get_reg(&mut cursor, UNW_REG_SP, &mut sp) != 0 {
+                writeln!(w, "{{ \"error\": \"Failed to retrieve registers during stack unwinding\" }}")?;
+                break;
+            }
+
+            if resolve_symbols {
+                let mut func_name = "<unknown>".to_string();
```

Comment:
does this have to be a string? That risks deadlocking on allocation inside a signal handler